app/controllers/master_files_controller.rb

```diff
@@ -170,8 +170,12 @@ def create
           # Use the errors key to signal that it should be a red notice box rather
           # than the default
           flash[:error] = "The file you have uploaded is too large"
-          redirect_to :back
-          return
+          return redirect_to :back
+        end
+
+        unless file.filename.valid_encoding? && file.name.is_ascii?
+          flash[:error] = 'The file you has non ASCII characters in its name.'
```

Comment:
Is "The file you have uploaded has non-ASCII characters in its name.' not long?  That would make it consistent with the too large message.